File: src/main/java/com/credowolf/depe/ui/modulesdialog/ModulesDialogForm.java
package com.credowolf.depe.ui.modulesdialog;

import com.intellij.openapi.module.Module;

import javax.swing.*;
import java.util.List;

public class ModulesDialogForm extends JPanel {
    JPanel contentPane;
    JRadioButton allModulesRadioButton;
    JRadioButton singleModuleRadioButton;
    JComboBox<String> modulesCombo;

    ModulesDialogForm(List<Module> modules) {
        ButtonGroup group = new ButtonGroup();
        group.add(allModulesRadioButton);
        group.add(singleModuleRadioButton);
        allModulesRadioButton.setSelected(true);
        for (Module module : modules) {
            modulesCombo.addItem(module.getName());
        }
        add(contentPane);
    }
}

File: src/main/java/com/credowolf/depe/ui/modulesdialog/ModulesDialogForm.kt
package com.credowolf.depe.ui.modulesdialog

import com.credowolf.depe.utils.*
import com.intellij.openapi.fileChooser.FileChooser
import com.intellij.openapi.project.Project
import com.intellij.openapi.project.guessProjectDir

import javax.swing.*

class ModulesDialogForm internal constructor(project: Project) : JPanel() {
    internal var contentPane: JPanel? = null
    internal var allModulesRadioButton: JRadioButton? = null
    internal var singleModuleRadioButton: JRadioButton? = null
    internal var modulesCombo: JComboBox<String>? = null
    internal var versionsFileName: JTextField? = null
    internal var selectFile: JButton? = null
    private var versionsFileLabel: JLabel? = null

    init {
        ButtonGroup().let {
            it.add(allModulesRadioButton)
            it.add(singleModuleRadioButton)
        }

        allModulesRadioButton!!.isSelected = true

        project.moduleManager.activeSubmodules.forEach {
            modulesCombo!!.addItem(it.name)
        }

        versionsFileName!!.text = project.getVersionsFileName()
        versionsFileName!!.isEditable = true

        selectFile!!.addActionListener {
            val projectDir = project.guessProjectDir()
            if(projectDir != null) {
                val virtualFile = FileChooser.chooseFile(createSingleGradleFileDescriptor(), project, project.guessVersionsFile())
                if (virtualFile != null) {
                    versionsFileName!!.text = virtualFile.path.removePrefix("${projectDir.path}/")
                }
            }
            else{
                //show error -> can't guess project dir
            }
        }

        add(contentPane)
    }
}
